source/extensions/filters/http/proto_message_logging/extractor_impl.cc

```diff
@@ -0,0 +1,123 @@
+#include "source/extensions/filters/http/proto_message_logging/extractor_impl.h"
+
+#include <string>
+#include <vector>
+
+#include "envoy/extensions/filters/http/proto_message_logging/v3/config.pb.h"
+#include "envoy/extensions/filters/http/proto_message_logging/v3/config.pb.validate.h"
+
+#include "source/extensions/filters/http/proto_message_logging/extractor.h"
+#include "source/extensions/filters/http/proto_message_logging/logging_util/proto_scrubber_interface.h"
+
+#include "absl/base/attributes.h"
+#include "absl/log/check.h"
+#include "absl/log/log.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "grpc_transcoding/type_helper.h"
+#include "proto_field_extraction/field_value_extractor/field_value_extractor_factory.h"
+#include "proto_field_extraction/field_value_extractor/field_value_extractor_interface.h"
+#include "proto_field_extraction/message_data/message_data.h"
+#include "proto_processing_lib/proto_scrubber/proto_scrubber_enums.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ProtoMessageLogging {
+namespace {
+
+using ::envoy::extensions::filters::http::proto_message_logging::v3::MethodLogging;
+using ::Envoy::Extensions::HttpFilters::ProtoMessageLogging::AuditDirective;
+using ::Envoy::Extensions::HttpFilters::ProtoMessageLogging::AuditMetadata;
+using ::google::grpc::transcoding::TypeHelper;
+using ::proto_processing_lib::proto_scrubber::ScrubberContext;
+using Protobuf::field_extraction::FieldValueExtractorFactory;
+
+// The type property value that will be included into the converted Struct.
+constexpr char kTypeProperty[] = "@type";
+
+ABSL_CONST_INIT const char* const kTypeServiceBaseUrl = "type.googleapis.com";
+
+void Extract(ProtoScrubberInterface& scrubber, Protobuf::field_extraction::MessageData& message,
+             std::vector<AuditMetadata>& vect) {
```

Comment:
Can it be called ScrubbedMessageMetadata etc? I think later it can be used for other API transformation and API filtering?

Reply:
Yes, right. Done.